### ISSUE TITLE
Reconfiguration fails if change source is polling

### ISSUE THREAD

**Issue by jugglinmike** [buildbot/buildbot#4030]
Reconfiguration will time out if the GitPoller (and likely other change sources) is polling for longer than 10 seconds. This may also apply to other change sources.

Steps to reproduce:

1. Configure a build master with the following change source:

   ```python
   c['change_source'] = [
       changes.GitPoller('git://github.com/w3c/web-platform-tests.git',
                         workdir='gitpoller-workdir',
                         branch='master',
                         pollAtLaunch=True,
                         pollInterval=300)
   ]
   ```
2. Create the build master and attempt to reconfigure it immediately:

       $ buildbot create-master master
       $ buildbot --verbose reconfig master

Output:

    2018-04-05 23:30:45+0000 [-] Log opened.
    2018-04-05 23:30:45+0000 [-] sending SIGHUP to process 2738
    2018-04-05 23:30:45+0000 [-] 2018-04-05 23:30:45+0000 [-] beginning configuration update
    2018-04-05 23:30:45+0000 [-] 2018-04-05 23:30:45+0000 [-] Loading configuration from '/home/user/master/master.cfg'
    2018-04-05 23:30:55+0000 [-] 2018-04-05 23:30:55+0000 [-] reconfig is ongoing for 10 s
    2018-04-05 23:31:05+0000 [-] Never saw reconfiguration finish.
    2018-04-05 23:31:05+0000 [-] Main loop terminated.

For the time being, I am working around this by scripting limited "retry" logic around the invocation of `buildbot reconfig`.

**Comment by thendrix** [comment]
I'm seeing this happen in master.cfg with a lot of logic as well since upgrading past 2.0.0.  If the config takes more than 10s to build it emits an error return code to the shell.  It's like a race condition if "Never saw reconfiguration finish." will happen or not.  To be clear the reconfig will finish successfully, however the CLI process has already exited by then.

When I get some free time I'm going to see if it's easy to modify master.py to simply adjust timing to avoid this.  My buildbots have CI running on master.cfg to update themselves, so this is annoying to have unstable return codes on CLI.

**Comment by thendrix** [comment]
I had to reapply this patch to 2.1 recently to WAR timeout causing partially applied reconfig.

**Comment by dankegel** [comment]
I'm getting this despite TIMEOUT_DELAY already being 30.

**Comment by p12tic** [comment]
I believe this issue has been fixed in #5795 and released as part of 2.10.2 and 3.0.0.
